peernode-market/src/producer/files.rs

```diff
@@ -28,6 +29,26 @@ pub fn hash_file(file: &mut File) -> Result<String> {
     let hash = format!("{:x}", hash);
     Ok(hash)
 }
+
+pub fn generate_chunk_metadata(file: &mut File) -> Result<Vec<(String, u64)>> {
+    let mut chunk_metadata = vec![];
+    
+    let mut sha256 = Sha256::new();
+    let mut buffer = vec![0; FileAccessType::CHUNK_SIZE as usize];
+    
+    loop {
+        let bytes_read = file.read(&mut buffer)?;
+        if bytes_read == 0 {
+            break;
+        }
+        sha256.update(&buffer[..bytes_read]);

```

Comment:
Reminder for me: 
Need to look into and clarify this in the Google Doc. Pretty sure using `update` here makes the hash of chunk `i` depend on the hash of `i-1`. If another team hashes each chunk with a new hasher we could get conflicting file keys.